uber/automated_emails.py

```diff
@@ -31,6 +31,21 @@ def __init__(self, model, subject, template, filter, *, sender=None, extra_data=
     def __repr__(self):
         return '<{}: {!r}>'.format(self.__class__.__name__, self.subject)
 
+    @property
+    def approved(self):
+        """Returns a boolean indicating whether this email has been approved."""
+        with Session() as session:
+            return bool(session.query(ApprovedEmail).filter_by(ident=self.ident).first())
+
+    def computed_subject(self, x):
+        """
+        Given a model instance, return an email subject email for that instance.
+        By default this just returns the default subject unmodified; this method
+        exists only to be overriden in subclasses.  For example, we might want
+        our panel email subjects to contain the name of the panel.
+        """
+        return self.subject
```

Comment:
Since it looks like this will get merged in advance of Dom's mega email refactor, I went ahead and added this utility method to the base ``AutomatedEmail`` class, since the panels department wants the ability to put panel names in their email subjects.